tests/unit/test_data_cleaning.py

```
@@ -0,0 +1,195 @@
+        assert mask.iloc[5] == True
+        assert not mask.iloc[0:5].any()
+        assert stats['method'] == 'iqr'
+        assert stats['outlier_count'] == 1
+
+    def test_detect_outliers_zscore(self):
+        """Test detect_outliers using Z-score method."""
+        # Using a larger set to have meaningful z-score
+        test_series = pd.Series([10, 10, 11, 10, 11, 10, 100])
+        mask, stats = detect_outliers(test_series, method='zscore', threshold=2.0)
+        
+        assert mask.iloc[6] == True
```

Ruff (0.14.13)

Avoid equality comparisons to `True`; use `mask.iloc[6]:` for truth checks

Replace with `mask.iloc[6]`

(E712)